virttest/qemu_vm.py

```diff
@@ -899,9 +899,9 @@ def add_memorys(devices, params):
             options, devs = [], []
             params.setdefault("mem", "512")
             mem_params = params.object_params("mem")
-            mem_params.setdefault("automem", "yes")
+            mem_params.setdefault("automem", "no")
             automem = mem_params["automem"] == "yes"
-            if automem:
+            if automem and mem_params.get("max_usable_mem"):
```

Comment:
evaluating memory should be a pre-step before create VM, so I think  env_process is a good place to do it, do it in qemu_vm.py will make qemu_vm module looks messy.